### PR TITLE
Add dex/CR/bad defaults failure story

### PR DESCRIPTION
**Pull request by pieterlange**
Slides for my failure story related to the [default](https://github.com/helm/charts/blob/master/stable/dex/values.yaml#L141) [dex](https://github.com/dexidp/dex/) configuration storing authrequests as CustomResources and its potential for nuking your kubernetes control plane.

The link: https://pieterlange.github.io/failure-stories/2019-06.dex.html
Ref: https://github.com/dexidp/dex/issues/1292
Shared at: https://www.meetup.com/Dutch-Kubernetes-Meetup/events/262313920/

### CODE REVIEW

**Comment by zerkms** [comment]
<details>
  <summary>Spoiler warning</summary>

> NO BUSINESS APPLICATIONS WERE HARMED DURING THIS OUTAGE.

</details>

^ that's truly impressive!

**Comment by pieterlange** [comment]
It was merely a (very scary!) control plane outage. The monitoring systems were running during the outage but inaccessible since dex (the auth system) was down so i did actually have the data to prove that the apps were up, after the storm was over.

The biggest challenge in recovering from this failure was doing so without access to the monitoring systems (at some point i actually did make a ssh portforward directly to the machines running the grafana/kibana pods). Fun times.

File: README.md [hunk 1]
```diff
@@ -129,6 +129,9 @@ Most recent publications on top.
 * [SaleMove US System Issue - SaleMove - incident report 2017](https://status.salemove.com/incidents/xf6cr710yrzn)
     * involved: AWS, ELB dynamic IPs, DNS `A` record for master, API server
     * impact: production issues with SaleMove US System
+* [A kubernetes failure story (dex) - Dutch kubernetes meetup slides 2019-06](https://pieterlange.github.io/failure-stories/2019-06.dex.html)
```
Comment (hjacobs) [inline README.md]: nit: kubernetes -> Kubernetes, company/org would be good in the title ({title} - {org} - {type} {year})

Comment (pieterlange) [reply README.md]: Cant add the actual company name but its one of our client platforms

**Comment by githubrotem** [comment]
Is there any real solution for this issue?
Right now anyone can make a curl loop and bring the cluster down

**Comment by szuecs** [comment]
We use skipper as kube-apiserver sidecar to do auth. and we can easily add client rate limits: 
https://opensource.zalando.com/skipper/reference/filters/#clientratelimit

Auth is done by tokens and validation is done by a tokeninfo sidecar that is fast enough.

**Comment by hjacobs** [comment]
@githubrotem see also https://twitter.com/pst418/status/1216739457400999938

**Pull request merged**